libraries/_helper.rb

```diff
@@ -433,12 +456,56 @@ def ensure_update_center_present!
         headers = {
           'Accept' => 'application/json'
         }
-        http = Net::HTTP.new(uri.host, uri.port)
-        http.use_ssl = true if uri.scheme == 'https'
-        response = http.post(uri.path, extracted_json, headers)
+        http = Chef::HTTP.new(uri)
+        http.post(uri, extracted_json, headers)
 
         true
       end
     end
+
+    #
+    # Idempotently create password for the keystore
+    #
+    def keystore_pass!
+      node.run_state[:jenkins_keystore_pass] ||= SecureRandom.uuid
+    end
+
+    #
+    # Idempotently create a java keystore from the CA certs in the trusted_cert_dir
+    #
+    def create_keystore!
+      node.run_state[:jenkins_create_keystore] ||= begin
+        FileUtils.rm_f(keystore)
+        certs = Dir.glob(File.join(Chef::Util::PathHelper.escape_glob(Chef::Config.trusted_certs_dir), '*.{crt,pem}'))
+        certs.each do |cert_file|
+          command = []
+          command << %Q(keytool)
```

Comment:
Yup. It should be calling the `keytool` method. Probably worked for the tests because `keytool` is was in the path.